lib/archethic/transaction_chain.ex

```diff
@@ -777,13 +777,18 @@ defmodule Archethic.TransactionChain do
   # ------------------------------------------------------------
 
   @doc """
-  Resolve all the last addresses from the transaction data
+  Resolve all the genesis addresses from the transaction data
+
+  **This function raises if it cannot fetch a genesis**
   """
-  @spec resolve_transaction_addresses(Transaction.t(), DateTime.t()) ::
+  @spec resolve_transaction_addresses(Transaction.t()) ::
```

Comment:
Suggestion: to keep the Elixir coding style, you could use the `!` naming for the function.